app/controllers/shelters_controller.rb

```diff
@@ -32,14 +31,29 @@ def update
   end
 
   def destroy
-    Shelter.destroy(params[:id])
+    shelter = Shelter.find(params[:id])
+    remove_favorited_pets(shelter)
+    shelter.destroy
     redirect_to '/shelters'
   end
 
   private
 
   def shelter_params
-    params.permit(:name, :address, :city, :state, :zip)
+    params.permit(:name,
+                  :address,
+                  :city,
+                  :state,
+                  :zip)
+  end
+
+  def remove_favorited_pets(shelter)
+    pets = shelter.pets.map { |pet| "#{pet.id}" }
+    pets.each do | pet_id |
+      if favorite.contents.include? pet_id
+        favorite.remove_pet(pet_id)
+      end
+    end
```

Comment:
And then re-write this as `pets.each {|pet_id| favorite.contents.delete(pet_id.to_s)}`